.jenkins/lm-eval-harness/test_lm_eval_correctness.py

```diff
@@ -59,24 +59,81 @@ def launch_lm_eval(eval_config):
     return results
 
 
-def report_performance(task, input_lens, output_lens, time):
+def report_performance(task, input_lens, output_lens, time, record_property):
     assert len(input_lens) == len(output_lens)
     context_lens = [i + o for i, o in zip(input_lens, output_lens)]
     gen_tput = sum(output_lens) / time
+    all_lens = [input_lens, output_lens, context_lens]
+    min_input_tokens, min_output_tokens, min_context_tokens = [
+        min(x) for x in all_lens
+    ]
+    max_input_tokens, max_output_tokens, max_context_tokens = [
+        max(x) for x in all_lens
+    ]
+    mean_input_tokens, mean_output_tokens, mean_context_tokens = [
+        statistics.mean(x) for x in all_lens
+    ]
+    stddev_input_tokens, stddev_output_tokens, stddev_context_tokens = [
+        statistics.stdev(x) for x in all_lens
+    ]
     msg = (
         f'{task} | estimated average generation throughput: {gen_tput:.2f} tokens/s \n'  # noqa: G004, E501
-        f'{task} | input_tokens   | min: {min(input_lens)} | max: {max(input_lens)} | mean: {statistics.mean(input_lens):.2f} | stddev: {statistics.stdev(input_lens):.2f}\n'  # noqa: E501
-        f'{task} | output_tokens  | min: {min(output_lens)} | max: {max(output_lens)} | mean: {statistics.mean(output_lens):.2f} | stddev: {statistics.stdev(output_lens):.2f}\n'  # noqa: E501
-        f'{task} | context_length | min: {min(context_lens)} | max: {max(context_lens)} | mean: {statistics.mean(context_lens):.2f} | stddev: {statistics.stdev(context_lens):.2f}'  # noqa: E501
+        f'{task} | input_tokens   | min: {min_input_tokens} | max: {max_input_tokens} | mean: {mean_input_tokens:.2f} | stddev: {stddev_input_tokens:.2f}\n'  # noqa: E501
+        f'{task} | output_tokens  | min: {min_output_tokens} | max: {max_output_tokens} | mean: {mean_output_tokens:.2f} | stddev: {stddev_output_tokens:.2f}\n'  # noqa: E501
+        f'{task} | context_length | min: {min_context_tokens} | max: {max_context_tokens} | mean: {mean_context_tokens:.2f} | stddev: {stddev_context_tokens:.2f}'  # noqa: E501
     )
+
+    # Log all of these stats to JUnitXML
+    record_property(f"{task}_gen_tput", gen_tput)
```

Comment:
It gives warnings in jenkins logs "PytestWarning: record_property is incompatible with junit_family 'xunit2' (use 'legacy' or 'xunit1')"
I think we can freely make it ignore it, so that logs are shorter:
-W 'ignore:record_property is incompatible with junit_family:pytest.PytestWarning'